pkg/report/table/vulnerability.go

```diff
@@ -175,33 +194,83 @@ func (r *vulnerabilityRenderer) printf(format string, args ...interface{}) {
 	_ = tml.Fprintf(r.w, format, args...)
 }
 
-func addParents(topItem treeprint.Tree, pkgID string, parentMap map[string][]string, seen map[string]struct{}) {
-	seen[pkgID] = struct{}{} // to avoid infinite loops
+func addParents(topItem treeprint.Tree, pkg ftypes.Package, parentMap map[string]ftypes.Packages, ancestors map[string][]string,
+	seen map[string]struct{}, depth int) {
+	if !pkg.Indirect {
+		return
+	}
 
-	for _, parent := range parentMap[pkgID] {
-		if _, ok := seen[parent]; ok {
-			return
+	roots := map[string]struct{}{}
+	for _, parent := range parentMap[pkg.ID] {
+		if _, ok := seen[parent.ID]; ok {
+			continue
+		}
+		seen[parent.ID] = struct{}{} // to avoid infinite loops
+
+		if depth < 1 {
+			branch := topItem.AddBranch(parent.ID)
+			addParents(branch, parent, parentMap, ancestors, seen, depth+1)
```

Comment:
You're right. I wanted to support `--dependency-depth` so the depth can be configured. But I eventually changed my mind because it could be too much at the moment. I'll remove it.

Reply:
Done https://github.com/aquasecurity/trivy/pull/3691/commits/02ba0cba4c01325abcec65b109a63857fbb80940